generate/generate.js

```diff
@@ -0,0 +1,253 @@
+const https = require("https");
+const fs = require("fs");
+
+// Get slug as argument
+const slug = process.argv[2];
+
+// Download the canonical data
+let dataUrl = `https://raw.githubusercontent.com/exercism/problem-specifications/main/exercises/${slug}/canonical-data.json`;
+https
+  .get(dataUrl, (res) => {
+    let body = "";
+    res.on("data", (chunk) => (body += chunk));
+    res.on("end", () => {
+      try {
+        let canonicalData = JSON.parse(body);
+        generateFiles(slug, canonicalData);
+      } catch (error) {
+        console.error(error.message);
+      }
+    });
+  })
+  .on("error", (error) => {
+    console.error(error.message);
+  });
+
+// Use the canonical data to generate the multiple files
+function generateFiles(slug, canonicalData) {
+  const exercise = kebabToPascal(slug);
+
+  let extractedFunctions = {};
+  extractFunctions(canonicalData, extractedFunctions);
+
+  const functionList = Object.entries(extractedFunctions)
+    .map(([key, _]) => key)
+    .join(", ");
+
+  const allFunctionsCode = Object.entries(extractedFunctions)
+    .map(([key, value]) => generateFunctionCode(key, value))
+    .join("\n\n");
+
+  const mainFile = `
+module ${exercise} exposing (${functionList})
+
+${allFunctionsCode}
```

Comment:
One way your script is superior to mine is that it keeps the order of the functions as they appear in the canonical data (I think). I store mine in a `Dict`, so the are printed in alphabetical order.

Reply:
Ah I had not thought of this. Keeping the order is rather interesting because usually the order makes sense and we want to keep it.